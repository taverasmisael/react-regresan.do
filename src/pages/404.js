import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import Button from 'material-ui/Button'

export default class NotFoundPage extends PureComponent {
  static propTypes = {
    prop: PropTypes.any
  }

  goLogin = () => this.props.history.push('/login')

  render() {
    return (
      <React.Fragment>
        <h1>Not Found</h1>
        <Button onClick={this.goLogin} raised color="primary">
          Login
        </Button>
      </React.Fragment>
    )
  }
}
