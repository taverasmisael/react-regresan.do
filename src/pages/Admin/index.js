import OverviewPage from './Overview'
import BranchesPage from './Branches'
import AdminPage from './Admin'

export { OverviewPage, BranchesPage, AdminPage }
