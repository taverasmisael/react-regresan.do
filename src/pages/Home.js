import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import Typography from 'material-ui/Typography'
export default class HomePage extends PureComponent {
  static propTypes = {
    prop: PropTypes.any
  }

  render() {
    return <Typography type="body1">Home</Typography>
  }
}
