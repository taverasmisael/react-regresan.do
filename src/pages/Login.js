import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import Typography from 'material-ui/Typography'
export default class LoginPage extends PureComponent {
  static propTypes = {
    prop: PropTypes.any
  }

  render() {
    return <Typography type="body1">Login</Typography>
  }
}
