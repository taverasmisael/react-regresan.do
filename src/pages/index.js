import HomePage from './Home'
import LoginPage from './Login'
import { AdminPage } from './Admin'

export { HomePage, AdminPage, LoginPage }
