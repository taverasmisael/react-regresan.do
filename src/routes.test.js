import React from 'react'
import { MemoryRouter } from 'react-router-dom'

import { shallow, mount } from 'enzyme'

import Routes from './routes'
import { NotFoundPage, HomePage, LoginPage } from './pages'

describe('routes', () => {
  describe('rendering', () => {
    it('renders without crashing', () => {
      shallow(<Routes />)
    })
    it('matches snapshot', () => {
      expect(shallow(<Routes />)).toMatchSnapshot()
    })
  })

  describe('functionality', () => {
    it('redirects to 404 if not matching route is found', () => {
      const component = mount(
        <MemoryRouter initialEntries={['/fakepath']}>
          <Routes />
        </MemoryRouter>
      )
      expect(component.find(NotFoundPage)).toHaveLength(1)
    })
    it('renders "HOME"', () => {
      const component = mount(
        <MemoryRouter initialEntries={['/']}>
          <Routes />
        </MemoryRouter>
      )
      expect(component.find(HomePage)).toHaveLength(1)
    })
    it('renders "Login"', () => {
      const component = mount(
        <MemoryRouter initialEntries={['/login']}>
          <Routes />
        </MemoryRouter>
      )
      expect(component.find(LoginPage)).toHaveLength(1)
    })
    it('does not renders "Admin" cause canActivate returns false', () => {
      const component = mount(
        <MemoryRouter initialEntries={['/admin']}>
          <Routes />
        </MemoryRouter>
      )
      expect(component.find(LoginPage)).toHaveLength(1)
    })
  })
})
