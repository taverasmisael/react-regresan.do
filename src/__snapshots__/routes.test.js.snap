// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes rendering matches snapshot 1`] = `
<Component>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <PrivateRoute
      canActivate={[Function]}
      component={[Function]}
      path="/admin"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</Component>
`;
